### PR TITLE
remembering the selection of mouse tool

### PR DESCRIPTION
**Pull request by moontrip**
This addresses https://github.com/VEuPathDB/web-eda/issues/971 with corresponding web-eda PR, https://github.com/VEuPathDB/web-eda/pull/1141.

Map related stories are updated as well.

### CODE REVIEW

**Review by bobular — CHANGES_REQUESTED** [review]
I was going to suggest that you export a MouseModes type from MouseTools, but I see you already do!!

Please could you import that into the Stories (and perhaps also export a `const defaultMouseMode : MouseMode = 'default'` so you can say

```
  const [mouseMode, setMouseMode] = useState<MouseMode>(defaultMouseMode);
```
?

This would make it more future-proof to the addition of more modes!


I also recommend changing the `setMouseMode` property to `onMouseModeChange` to be more consistent (though I see we are not very consistent with the use of `Changed` and `Change`)
I notice on the EDA side you have called it `onMouseModeChange`

**Comment by moontrip** [comment]
@bobular Thank you for your review and suggestions! 👍 I have addressed your feedbacks.

File: src/map/MapVEuMap.tsx [hunk 1]
```diff
@@ -156,6 +156,10 @@ export interface MapVEuMapProps {
   showNoDataOverlay?: boolean;
   /** Whether to show the Scale in the map */
   showScale?: boolean;
+  /** mouseMode control */
+  mouseMode: 'default' | 'magnification';
```
Comment (bobular) [inline src/map/MapVEuMap.tsx:160]: We could save some typing/maintenance with
```suggestion
  mouseMode: MouseMode;
```

File: src/map/MapVEuMap.tsx [hunk 2]
```diff
@@ -156,6 +156,10 @@ export interface MapVEuMapProps {
...
+  /** a function for changing mouseMode */
+  onMouseModeChange: (value: 'default' | 'magnification') => void;
```
Comment (bobular) [inline src/map/MapVEuMap.tsx:162]:
```suggestion
  onMouseModeChange: (value: MouseMode) => void;
```

**Review by bobular — APPROVED** [review]
Looks great, but see my suggestions :-)

**Pull request merged**